File: LeetCode/530MinimumAbsoluteDifferenceInBST/MinimumAbsoluteDifferenceInBST.cpp
class Solution {
public:
    int minDiff = INT_MAX;
    void inorder(TreeNode* root, TreeNode* &prev) {
        if (!root) {
            return;
        }
        inorder(root->left, prev);
        if (prev) {
            minDiff = min(minDiff, root->val - prev->val);
        }
        prev = root;
        inorder(root->right, prev);
    }
    int getMinimumDifference(TreeNode* root) {
        // Inorder -> Time:O(n), Space:O(h)
        TreeNode* prev = NULL;
        inorder(root, prev);
        return minDiff;
    }
};

File: LeetCode/530MinimumAbsoluteDifferenceInBST/MinimumAbsoluteDifferenceInBST.py
class Solution:
    min_diff = float("inf")
    prev = None

    def dfs(self, root):
        if not root:
            return
        self.dfs(root.left)
        if self.prev:
            self.min_diff = min(self.min_diff, root.val - self.prev.val)
        self.prev = root
        self.dfs(root.right)

    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        # Inorder -> Time:O(n), Space:O(h)
        self.dfs(root)
        return self.min_diff
